File: kinoafisha/Legacy/Model/Film.m
//
//  Film.m
//  kinoafisha
//
//  Created by Eugene Dorfman on 11/11/14.
//  Copyright (c) 2014 justadreamer. All rights reserved.
//

#import "Film.h"
#import "Attribute.h"
#import "ScheduleEntry.h"

@implementation Film
+ (NSDictionary *)JSONKeyPathsByPropertyKey {
    return @{
        @"title":@"title",
        @"subtitle":@"subtitle",
        @"thumbnailURL":@"thumbnail",
        @"rating":@"rating",
        @"votesCount":@"votes_count",
        @"attributes":@"attributes",
        @"descr":@"description",
        @"scheduleEntries":@"schedule",
        @"detailURL":@"detail_url"
        };
}

+ (NSValueTransformer *)thumbnailURLJSONTransformer {
    return [NSValueTransformer valueTransformerForName:MTLURLValueTransformerName];
}

+ (NSValueTransformer *)attributesJSONTransformer {
    return [MTLJSONAdapter arrayTransformerWithModelClass:Attribute.class];
}

+ (NSValueTransformer *)scheduleEntriesJSONTransformer {
    return [MTLJSONAdapter arrayTransformerWithModelClass:ScheduleEntry.class];
}

+ (NSValueTransformer *)detailURLJSONTransformer {
    return [NSValueTransformer valueTransformerForName:MTLURLValueTransformerName];
}

@end

File: kinoafisha/Films/Film.swift
//
//  Film.swift
//  kinoafisha
//
//  Created by Eugene Dorfman on 13/7/19.
//  Copyright © 2019 justadreamer. All rights reserved.
//

import Foundation

struct Film: Codable, Equatable {
    enum CodingKeys: String, CodingKey {
        case title
        case subtitle
        case thumbnailURL = "thumbnail"
        case rating
        case votesCount = "votes_count"
        case attributes
        case descr = "description"
        case scheduleEntries = "schedule"
        //case detailURL = "detail_url"
        case detailParsedRequest = "detail_parsed_request"
    }
    var title: String
    var subtitle: String?
    var thumbnailURL: URL?
    var detailParsedRequest: ParsedRequest?
    var rating: String
    var votesCount: String
    var attributes: [Attribute]
    var descr: String?
    var scheduleEntries: [ScheduleEntry]?
}
